sql/core/src/test/scala/org/apache/spark/sql/streaming/continuous/ContinuousSuite.scala

```diff
@@ -38,18 +38,42 @@ class ContinuousSuiteBase extends StreamTest {
       sparkConf.set("spark.sql.testkey", "true")))
 
   protected def waitForRateSourceTriggers(query: StreamExecution, numTriggers: Int): Unit = {
-    query match {
+    findRateStreamContinuousStream(query).foreach { reader =>
+      val deltaMs = numTriggers * 1000 + 300
+      val firstCommittedTime = reader.firstCommittedTime.longValue()
```

Comment:
This change is based on the observation: while `reader.creationTime` doesn't guarantee partition readers are initialized. Instead, `reader.firstCommittedTime` guarantees partition readers are initialized before, so we can capture and use this.
(This is a best-effort on driver side, not fastest approach, of course.)